v2/client.go

```diff
@@ -41,6 +42,9 @@ func NewClient(config *ClientConfiguration) (Client, error) {
 	if config.Insecure {
```

Comment:
we might consider a validation on the config.  The http library afaik will not let you set insecure-skip-verify and use RootCAs at the same time.